circuit-test/src/commonMain/kotlin/com/slack/circuit/test/PresenterTestExtensions.kt

```diff
@@ -28,9 +32,10 @@ public suspend fun <UiState : CircuitUiState> Presenter<UiState>.test(
   timeout: Duration? = null,
   name: String? = null,
   policy: SnapshotMutationPolicy<UiState> = structuralEqualityPolicy(),
+  retainedStateRegistry: RetainedStateRegistry = RetainedStateRegistry(),
```

Comment:
WDYT to making this nullable (and default to null)?

Providing a default `retainedStateRegistry` is a bit of a foot gun, as it's so easy for devs to ignore it. At least if we force people to provide a value, it's on them to track.

Reply:
I'd be game for that or default to NoopStateRegistry, though I'm still not sure this API is necessary to expose at the top level as the only use case right now was automatic-cleanup.

Reply:
Yeah, this is basically just recreating `NoopStateRegistry`. Maybe this PR isn't needed, if people want to provide their a real registry in tests then they're free to do so. Might be worth a doc call out though.